Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -11,141 +11,197 @@
+            return getConfig().getBoolean(key, defaultValue);
         } catch (ConversionException e) {
-            return defaultIfNull;
+            logger.catching(e);
+            logger.warn(
+                "Configuration found in kitodo_config.properties for key {} is defined, but cannot be converted to boolean!",
+                key);
+            return defaultValue;
         }
     }
 
     /**
-     * Request int-parameter from Configuration.
+     * Returns the selected int parameter from the configuration file. If no
+     * such parameter exists, or the value cannot be parsed to {@code int},
+     * returns {@code 0}.
      *
-     * @param inParameter
-     *            name of parameter in config file
-     * @return Parameter as int
+     * @param key
+     *            key whose value is to be returned
+     * @return int value for the requested key, or {@code 0} if not found or not
+     *         parsing
      */
-    public static int getIntParameter(String inParameter) {
-        return getIntParameter(inParameter, 0);
+    public static int getIntParameter(String key) {
+        return getIntParameter(key, INT_PARAMETER_NOT_DEFINED_OR_ERRONEOUS);
     }
 
     /**
-     * Request int-parameter from Configuration with default-value.
+     * Returns the selected int parameter from the configuration file. If no
+     * such parameter exists, or the value cannot be parsed to {@code int},
+     * returns the provided default value.
      *
-     * @param inParameter
-     *            name of parameter in config file
-     * @param inDefault
-     *            default value in case parameter taken from config file is null or
-     *            exception occurred
-     * @return Parameter as int
+     * @param parameter
+     *            key whose value is to be returned
+     * @param defaultIfNull
+     *            default value in case parameter taken from config file does
+     *            not exist or exception occurred
+     * @return int value for the requested key, or {@code defaultIfNull} if not
+     *         found or not parsing
      */
-    public static int getIntParameter(String inParameter, int inDefault) {
+    public static int getIntParameter(String key, int defaultValue) {
         try {
-            return getConfig().getInt(inParameter, inDefault);
+            return getConfig().getInt(key, defaultValue);
         } catch (ConversionException e) {
-            return inDefault;
+            logger.catching(e);
+            logger.warn(
+                "Configuration found in kitodo_config.properties for key {} is defined, but cannot be converted to int!",
+                key);
+            return defaultValue;
         }
     }
 
     /**
-     * Gets the configuration.
+     * Returns the configuration.
      *
-     * @return the PropertyConfiguration
+     * @return the configuration
      */
     public static PropertiesConfiguration getConfig() {
         if (config == null) {
```

Comment:
I think this method needs some deeper analyze.

```
if (config == null) {
PropertiesConfiguration initialized = config;
if (initialized == null) {
```

The null value in second if is obvious.

Reply:
No. This is a concurrency issue and a very hairy subject. There is [a long explaination on Wikipedia covering why this is necessary](https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java).